File: backend/app/routers/rules.py
from fastapi import APIRouter
from pathlib import Path
import json

router = APIRouter(tags=['rules'])

DATA_DIR = Path(__file__).parent.parent / 'data'


@router.get('/rules')
def get_rules():
    data = json.loads((DATA_DIR / 'rules.json').read_text(encoding='utf-8'))
    return data

File: backend/src/routers/rules.ts
import { Router } from 'express'
import { readFileSync } from 'fs'
import { fileURLToPath } from 'url'
import { dirname, join } from 'path'

const __dirname = dirname(fileURLToPath(import.meta.url))
const DATA_DIR = join(__dirname, '..', 'data')

const router = Router()

router.get('/rules', (_req, res) => {
  const data = JSON.parse(readFileSync(join(DATA_DIR, 'rules.json'), 'utf-8'))
  res.json(data)
})

export default router
